graph/pkg/service/v0/drives.go

```diff
@@ -382,12 +382,15 @@ func (g Graph) UpdateDrive(w http.ResponseWriter, r *http.Request) {
 func (g Graph) formatDrives(ctx context.Context, baseURL *url.URL, storageSpaces []*storageprovider.StorageSpace) ([]*libregraph.Drive, error) {
 	responses := make([]*libregraph.Drive, 0, len(storageSpaces))
 	for _, storageSpace := range storageSpaces {
-		res, err := g.cs3StorageSpaceToDrive(baseURL, storageSpace)
+		res, err := g.cs3StorageSpaceToDrive(ctx, baseURL, storageSpace)
 		if err != nil {
 			return nil, err
 		}
 		res.Special = g.GetExtendedSpaceProperties(ctx, baseURL, storageSpace)
 		res.Quota, err = g.getDriveQuota(ctx, storageSpace)
+		if res.Root.RemoteItem != nil {
+			res.Root.RemoteItem, err = g.getRemoteItem(ctx, resourceid.OwnCloudResourceIDUnwrap(*res.Root.RemoteItem.Id), baseURL)
+		}
```

Comment:
👀

Reply:
I don't get that. What is happening here? Why are we overwriting the `RemoteItem`?

Reply:
RemoteItem in the response is not always set. When we have a RemoteItem in the response, then we only have the ResourceID. Therefore we need to stat the resourceID in `g.getRemoteItem` and get all the properties. Then we replace the remoteItem with the result from the stat.

Reply:
Ah, would be good to have that commented. I can see somebody in the future removing this block because it's not immediately clear.

Reply:
But we should untangle this.

Reply:
I will find a better solution